src/ax/saga/workflow.go

```diff
@@ -14,75 +15,122 @@ type Workflow struct {
 	ErrorIfNotFound
 	CompletableByData
 
-	Prototype   Data
-	Triggers    ax.MessageTypeSet
-	NonTriggers ax.MessageTypeSet
-	Handle      typeswitch.Switch
+	Prototype     Data
+	Triggers      ax.MessageTypeSet
+	NonTriggers   ax.MessageTypeSet
+	CommandHandle typeswitch.Switch
+	EventHandle   typeswitch.Switch
 }
 
 // NewWorkflow returns a saga that forwards to the given aggregate.
 //
-// Workflows are a specialization of sagas that handle events and produce
-// commands.
+// Workflows are a specialization of sagas that handle commands and/or events
+// and produce commands.
 //
 // It accepts a prototype data instance which is cloned for new instances.
 //
-// For each event type to be handled, the aggregate must implement a "handler"
+// For each message type to be handled, the aggregate must implement a "handler"
 // method that adheres to one of the following signatures:
 //
-//     func (ev *<T>, ax.CommandExecutor)
-//     func (ev *<T>, env ax.Envelope, ax.CommandExecutor)
+//     func (m *<T>, ax.CommandExecutor)
+//     func (m *<T>, env ax.Envelope, ax.CommandExecutor)
 //
-// Where T is a struct type that implements ax.Event.
+// Where T is a struct type that implements ax.Message.
 //
 // Handler methods are responsible for mutating the state of the workflow and
-// producing new commands, based on the event being handled. They may inspect
-// the current state of the workflow, and then return zero or more commands
-// to be executed.
+// producing new commands, based on the message being handled.
 //
-// The names of handler methods are meaningful to the workflow system. If an
-// event is meant to trigger a new workflow instance, its handler method's name
-// must begin with "StartWhen". Other handler methods must begin with "When". By
-// convention these prefixes are followed by the message name, such as:
+// The names of handler methods are meaningful to the workflow system. If a
+// command or event is meant to trigger a new workflow instance, its handler
+// method's name must have a prefix "Begin". Non-triggering command handler
+// methods must have prefix "Do". Both triggering and non-triggering event
+// handler methods must have prefix "When". For triggering event handlers the
+// combined prefix results in "BeginWhen". By convention these prefixes are
+// followed by the message name, such as:
```

Comment:
This section is a bit hard to explain, I had a go at rewording it:

> The names of handler methods are meaningful to the workflow system. If a message is meant to trigger a new workflow instance, its handler method's name must prefixed with "Begin", if it is a command handler, or "BeginWhen" if it is an event handler. Messages that can be routed to existing workflow instances, but not cause new instances must have their method names prefixed with "Do" and
"When" for commands and events, respectively. 
>
> By convention these prefixes are followed by the message name, such as: